oscarlab/graphene
# [Pal] Clean up GDB configuration

See #1814 for discussion of specific GDB options.

## Description of the changes 

- Extract parts that are common for all hosts
- Remove some outdated/unnecessary options, we should now be
  closer to default configuration
- Disable libthread_db loading (does not work and crashes GDB 9.2)
- Disable pagination when loading debug maps   

## How to test this PR? 

* Try running some programs with GDB (under Linux and Linux-SGX).

* There are some smoke tests for GDB integration in `LibOS/shim/test/regression` (runnable with `~/graphene/Scripts/run-pytest -k TC_50_GDB`).

* Install with Meson and check if the `gdb_integration/common` directory is copied over properly:

```sh
meson build -Ddirect=enabled
DESTDIR=$PWD/install ninja -C build install
find install/
```


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/oscarlab/graphene/2129)
